api/HTMLButtonElement.json

```diff
@@ -942,10 +942,10 @@
               "version_added": "≤12.1"
             },
             "safari": {
-              "version_added": "3.2"
+              "version_added": "4"
```

Comment:
Quoting from https://github.com/mdn/browser-compat-data/pull/7272/files#r517971169:

@foolip:

> How confident are you that this wasn't supported in Safari 3? Which platform was Safari 3 tested on?
>
> https://trac.webkit.org/changeset/35381/webkit added the `willValidate` attribute to a bunch of interfaces, and the timing of it is consistent with is being in Safari 3.2 and Safari for iOS 3.

@vinyldarkscratch:

> I tested these in Mac OS X on Safari 3.2.3 (which is the latest 3.2 release).

http://mdn-bcd-collector.appspot.com/tests/api/HTMLButtonElement/willValidate looks good, so I don't think the test is at fault.

Timing of the change aside, it looks like this was simply after the Safari 3.2 branch, it shows up between Safari 3.2.1 and Safari 4.0:
https://trac.webkit.org/browser/webkit/releases/Apple/Safari%203.2.1/WebCore/html/HTMLButtonElement.idl
https://trac.webkit.org/browser/webkit/releases/Apple/Safari%204.0/WebCore/html/HTMLButtonElement.idl

It's excellent that our tests are turning out to be correct instead of our inferences from source code! :D